src/mapper.rs

```diff
@@ -296,8 +321,8 @@ impl<'s> ProguardMapper<'s> {
 
     /// returns a tuple where the first element is the list of the function
     /// parameters and the second one is the return type
-    pub fn deobfuscate_signature(&'s self, signature: &str) -> Option<(Vec<String>, String)> {
-        java::deobfuscate_bytecode_signature(signature, self)
+    pub fn deobfuscate_signature(&'s self, signature: &str) -> Option<DeobfuscatedSignature> {
+        DeobfuscatedSignature::new(java::deobfuscate_bytecode_signature(signature, self))
```

Comment:
```suggestion
        java::deobfuscate_bytecode_signature(signature, self).map(DeobfuscatedSignature::new)
```
Assuming you remove the `map` and `Option`s from the constructor.